src/dev/flang/ast/Destructure.java

```diff
@@ -217,6 +217,14 @@ private void addAssign(Resolution res,
     if (fields != null && fields.hasNext())
       {
         var newF = (Feature) fields.next();
+
+        var a = newF._returnType.functionReturnType().resolve(res, context);
+
+        if (a != Types.t_UNDEFINED && !(!newF._returnType.isConstructorType() && a.isAssignableFrom(t)) )
+          {
```

Comment:
This looks dangerous and too complex. `newF._returnType.functionReturnType()` should crash if `newF._returnType.isConstructorType()`, which is checked afterwards but with a `!(!(` that asks for restructuring.